### DISCUSSION TITLE
How to import markdown files

### DISCUSSION THREAD

**Discussion by shinokada** [fastapi/fastapi#7100] [Questions] [answered]
What is the best way to import a markdown file?
I have markdown files in the `app/pages/` directory and want to import them.

I'd like to use these markdown files for frontend texts with using [Python-Markdown](https://pypi.org/project/Markdown/).

I tried this but didn't work:

```
readme_file = open("./pages/about.md", "r")
md_template_string = markdown.markdown(
        readme_file.read()
    )
```

**Comment by Kludex** [comment]
If this is related to FastAPI, you probably need to give more context here... 

Is this useful? https://python-markdown.github.io/reference/#markdown

**Comment by shinokada** [comment]
I used it in this way:

```
@app.get("/markdown", response_class=HTMLResponse)
async def color_home(request: Request):
    with open("./pages/about.md", "r", encoding="utf-8") as input_file:
        text = input_file.read()
    html = markdown.markdown(text)

    data = {
        "title": "About Artist Tools",
        "text": html
    }

    return templates.TemplateResponse("page.html", {"request": request, "data": data})
```


And I get the following error:

> with open("./pages/about.md", "r", encoding="utf-8") as input_file:
> FileNotFoundError: [Errno 2] No such file or directory: './pages/about.md'

**Comment by ycd** [comment, marked as answer]
@shinokada well this is jıst a `FileNotFoundError`. 

Change the path to `app/pages/about.md` I'm not sure though but you need to change the path.

**Comment by shinokada** [comment]
@ycd 
Thank it worked.

**Comment by tiangolo** [comment]
Thanks for the help here everyone! :clap: :bow: 

Thanks for reporting back and closing the issue :+1:
